### ISSUE TITLE
Webpacker, Vue JS and serving from subfolder - Rails 5.2

### ISSUE THREAD

**Issue by luigi7up** [rails/webpacker#2907]
Hi guys,

I know this might be a Vue issue, but maybe, just maybe someone might help me here :/

I'm trying to move a Vue 3 app that's working fine when setup through the CLI and ran using the `npm run serve`

The thing is that I want to serve this app as a part of an existing Rails 5 app. So, I installed the Webpacker, set it up, used the `javascript_pack_tag()` and got to render my Vue components correctly.

The problem is that I want to serve the Vue not in the root '/', but in a subpath '/spa/'

Following the Vue docs I added the `base` param into the Vue router:

```
const routes = [
  {
    path: "/q1",
    name: "q1",
    component: Question1,
  },
  {
    path: "/q2",
    name: "q2",
    component: Question2,
  },
  { 
    path: "/", 
    redirect: { name: 'q1' }
  }
];

const router = createRouter({
  base: '/spa/',
  mode:history,
  history: createWebHistory(),
  routes,
});

export default router;
```

When I visit `localhost:3000/spa` I can see my Vue components compiled, but the console says that there is no path corresponding to `/spa`. If I click on a +<router-link to="/q1" />` the components gets correctly rendered, but the URL is not `/spa/q1`, but `/q1`🤔

Does anyone know how to tell Vue that the initialization path of the Vue app is `/spa/` ant that that's it's root path?

I'm not sure whether this is a Vue issue, or Webpack/Webpacker issue 

Thank you so much

**Comment by pedrofurtado** [comment]
Hey, @luigi7up ! 👋 

Could you give us more details about your app? For example: are you using webpacker 5.x version, with rails 5.x and vue 3.x? Do you have some reproducible repo? In order to allow us help you to find the root cause of this problem that you described 🔍 🕵️ 

🤝

**Comment by luigi7up** [comment]
@pedrofurtado HI! So sorry for not giving any update on this one. The problem was that I was using the new Vue Loader v3 syntax, but I had installed the Vue loader ~v4
- In vue-router 3.0, you instantiate a router via new Router. You provide a base option to the new Router instance.
- In vue-router 4.0, you do this via the createRouter function. The routerOptions no longer accept a base key; base is now an optional argument for the createHistory function.

So, I changed the 

```
const router = createRouter({
  history: createWebHistory(),
  base: '/spa/',
  routes,
});
```

to

const router = createRouter({
  mode:history,
  history: createWebHistory( '/spa/' ),
  routes,
});

and that was it...

Unfortunatelly Vue router doesn't complain whn you pass the `base: "xxxx"` property when initializing even though it's not expected there anymore
